### ISSUE TITLE
Sending both Command Ack + Telemetry causes "IoTConnect Client message send failed!" (At least on Azure)

### ISSUE THREAD

**Issue by akarnil** [avnet-iotconnect/iotc-generic-c-sdk#50]
I am connecting to the Azure instance.

Using the latest basic sample code with no changes apart from credentials, If I send a command to the device with an ack it tries to send a message back but sometimes it fails and can cause the entire application to crash or stop sending any telemetry.

The command is just a dummy one.

```
<: {"v":"2.1","ct":0,"cmd":"set_standard_price.sh 123123","ack":"2e7b3fd6-5c85-45a4-927a-2f3fc9523bca"}
Command set_standard_price.sh 123123 received with 2e7b3fd6-5c85-45a4-927a-2f3fc9523bca ACK ID
>: {"d":{"ack":"2e7b3fd6-5c85-45a4-927a-2f3fc9523bca","st":4,"type":0,"msg":"Not implemented"}}
>: {"d":[{"d":{"version":"00.01.00","random_int":2,"random_decimal":0.064423763223189753,"random_boolean":false,"coordinate":{"x":0.22862574096239441},"coordinate":{"y":3.2687862372346159}}}]}
IoTConnect Client message send failed!
```

**Comment by nik-markovic** [comment]
@akarnil Could it be possible that for some reason the topic where we need to send the ack is blocked by the back end because the topic not unautorized? Is this potentially some device that was created long ago in IoTConnect? If so, would you mind deleting and re-creating the device with same credentials (if applicable)?

**Comment by nik-markovic** [comment]
@alanlhc I did investigate the issue further and I found that if sending ACKs while riding on the messageArrived callback thread from the paho client, the waitForCompletion call would hang. for now I added a workaround to send acks with QOS 0 and that seems to clear it up. 

We should spend some time to investigate this further and do things better. I am pressed on time now, so I can't, but feel free to help out if you have time to spare.

Make sure to also pull in the latest clib with the git submodule command there after pulling the main repo as you will want to be picking up the change that sends dummy timestamp along witht the acks.
